nix/pkgs/ms-buildenv/default.nix

```diff
@@ -0,0 +1,201 @@
+#
+# When we use contents []; what in effect will happen is that during the
+# generation of the layers, /bin /include and /lib gets"flattend" into a
+# single directory. (lack of a better term sorry). What this means is that
+# if we have pkgs.A and pkgs.B it would yield, typically;
+#
+# /store/nix/<hash>-A/{bin,lib,include}
+# /store/nix/<hash>-B/{bin,lib,include}
+#
+# After we have a "contents" all outputs get stored in /include and /lib
+# this is fine, however, some of the team would like to have an actual nix
+# environment, that is to say, the nix tools and nix-env -q remain working.
+#
+# In the case of mayastor for example this is exactly what we want. We have
+# /bin/mayastor and a couple of /libs.
+#
+# To keep the existing derivations working, nix-style we must construct a
+# user profile as you normally would have if you where running nix.
+#
+# A user profile is a symlinked together directory /nix/var/nix/profiles/<name>
+# the construction of the profile is done by nix-env so we emulate that within
+# docker layer cake, by importing the output derivations in to the nix-store
+# after building the packages.
+#
+
+{ bashInteractive
+, binutils
+, cacert
+, coreutils
+, findutils
+, git
+, gnugrep
+, gnused
+, closureInfo
+, rustup
+, numactl
+, dockerTools
+, liburing
+, openssl
+, rdma-core
+, utillinux
+, libspdk
+, libiscsi
+, mkContainerEnv
+, libaio
+, binutils-unwrapped
+, nodejs
+, protobuf
+, llvmPackages
+, gzip
+, iproute
+, # vscode
```

Comment:
I think this comment should be removed. It is unlikely that anyone would like to run vscode editor inside the container. Instead it's just going to be used for remote executions.

Reply:
I'll move the references now but -- AFAIK there are two "modes" and if I where to use this stuff, i would run vscode *in* the container, and not SSH (i think...)